### DISCUSSION TITLE
Importing YAML Files

### DISCUSSION THREAD

**Discussion by tyteen4a03** [vercel/next.js#14102] [Help]
How do I import YAML files? When I follow this guide: https://github.com/vercel/next.js/issues/3141

I run into this issue:

```
wait  - compiling...
error - ./awards.yaml 3:10
Module parse failed: Unexpected token (3:10)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| dataVersion: 18042020
| years:
>     "2018": 1
|
```

When I try to `import awards from './awards.yaml';`

awards.yaml:

```
dataVersion: 18042020
years:
     "2018": 1
````

**Comment by tyteen4a03** [comment]
I ended up just importing the yaml file during `getStaticProps`. Not the most efficient method but works.

**Comment by frencojobs** [comment]
You can use [next-plugin-yaml](https://www.npmjs.com/package/next-plugin-yaml) as a plugin inside `next.config.js` without touching the webpack config. If you're using typescript, you might also need to declare a custom module for the file, so that typescript recognizes the import.
